### PR TITLE
Added testing procedure

### PR DESCRIPTION
**Pull request by ghost**
To solve #1.

A testing procedure with docker. The testing starts from a docker image with Ubuntu 18.04 and MATLAB. The procedure for generating such image can be found in the `test/0-matlab` directory of the PR.

### CODE REVIEW

**Comment by gilbertolem** [comment]
Up to now, the bindings are tested by running the solverdummy and two versions (explicit and implicit coupling) of a tutorial of a LC circuit. However this does not cover the full API, as mentioned in https://github.com/precice/matlab-bindings/pull/14#discussion_r374950634, and unit testing / mocking is necessary.

**Comment by BenjaminRueth** [comment]
Note that due to https://github.com/precice/systemtests/issues/102 we had to skip the testing when building the docker image. However, I think it would be better to just exclude the failing test (see [`precice.parallel`](https://travis-ci.org/precice/systemtests/jobs/657698774#L2552-L2553)) and run the other tests until https://github.com/precice/systemtests/issues/102 is resolved.

**Comment by fsimonis** [comment]
We dropped support for Ubuntu 18.04LTS could you migrate to 20.04LTS?

**Comment by erikscheurer** [comment]
The testing strategy will be completely changed to not use docker anymore. If we get [parallel execution of matlab instances](https://github.com/matlab-actions/run-command/issues/22) to work we will use the matlab-actions for github to setup matlab, build the bindings and run the solverdummies this way. 
This means we also won't use Ubuntu 18.04LTS

**Comment by erikscheurer** [comment]
Sorry for the spam of commits, I couldn't get it to work locally.
The current setup of actions is this:
- On `push` and `pull_request` the `test-solverdummies` and `test-tutorial` actions in [`test.yml`](https://github.com/precice/matlab-bindings/blob/b0da5090c8a1cfbd62a20e0074c9e5c9c8137138/.github/workflows/test.yml#L4) are triggered. 
- They make use of a [strategy matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) to test multiple releases of Matlab. Currently tested are versions: `'latest'`, `R2021b`, `R2021a`, `R2020b`, `R2020a`
- The test actions call a [composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) saved in [`action.yml`](https://github.com/precice/matlab-bindings/blob/testing/.github/workflows/action.yml) with the given matlab release and preCICE release. This composite action:
  - installs preCICE through the [installation package](https://github.com/precice/precice/releases)
  - installs matlab and compiles the preCICE bindings
  - can be reused by others in workflows that want to do automatic testing with the preCICE matlab-bindings
  - is called `action.yml`. Should it be moved into a subfolder `build/action.yml`? I didn't find a way to call the action build.yml and the action being recognized, does anyone know a way?
- After the bindings are set up, two instances of matlab are called simultaniously using the [bash subshells](https://stackoverflow.com/questions/5547787/running-shell-script-in-parallel) aka concatenating two instances of `matlab -batch` with `&` 

There is no action that only builds the bindings since it's required in both tests. If the build fails, the test executions don't trigger. You can see that the action fails in the `build bindings` action. Should a build action be executed isolated?
Github actions apparently is able to [reuse workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations) aka make a build workflow and go from there to the testing workflow to avoid building twice. However, this feature doesn't support a matrix, which disqualifies it in this case.

There still is no unit testing sadly, but I think this should be handled in a different issue and PR.

File: .github/workflows/action.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: 'Build Matlab bindings'
+description: 'Installs precice and builds the preCICE matlab bindings in the directory. Ends in `$GITHUB_WORKSPACE/matlab-bindings`'
```
Comment (IshaanDesai) [inline .github/workflows/action.yml:2]: preCICE* and either Matlab* or MATLAB* everywhere (please check the correct nomenclature)

**Comment by erikscheurer** [comment]
> I did not fully get the relevance of some of the files, do we really need the file `test/0-matlab/README` for example?

These files were part of the previous approach to testing. I deleted them now.

**Review by IshaanDesai — APPROVED** [review]
Lets merge this :+1:

**Pull request merged**